### PR TITLE
Add more metrics log

### LINKED ISSUE #491: Add more metrics for aggregating and cached blocks
**Describe what enhancement you'd like to have**
Record the metrics of some important nodes, for examples:
1) Aggregating
2) Join
3) Changelog Converting

### PR DESCRIPTION
**Pull request by yl-lisen**
This close #491 

Please write user-readable short description of the changes:
- Add aggregated data metrics
- Add more metrics for cached blocks
- Remove metrics serialization (we should avoid serialize/deserialize metrics, since they are volatile)

### CODE REVIEW

File: src/Core/Block.cpp [hunk 1]
```diff
@@ -397,6 +402,18 @@ size_t Block::allocatedBytes() const
     return res;
 }
 
+size_t Block::allocatedMetadataBytes() const
+{
+    size_t res = 0;
+    for (const auto & elem : data)
+        res += elem.column->allocatedMetadataBytes();
+
+    res += sizeof(data) + data.capacity() * sizeof(ColumnWithTypeAndName);
```
Comment (chenziliang) [inline src/Core/Block.cpp:411]: ColumnWithTypeAndName itself will allocate, we are counting short ?

File: src/Interpreters/Streaming/Aggregator.cpp [hunk 2]
```diff
@@ -142,6 +143,39 @@ void AggregatedDataVariants::convertToTwoLevel()
     }
 }
 
+void AggregatedDataVariants::updateMetrics(AggregatedDataMetrics & metrics) const
```
Comment (chenziliang) [inline src/Interpreters/Streaming/Aggregator.cpp]: How expensive is for this call ?

File: src/Interpreters/Streaming/CachedBlockMetrics.cpp [hunk 3]
```diff
@@ -7,22 +7,26 @@ namespace DB
 {
 namespace Streaming
 {
-void CachedBlockMetrics::serialize(WriteBuffer & wb) const
+void CachedBlockMetrics::serialize(WriteBuffer & wb, VersionType version) const
```
Comment (chenziliang) [inline src/Interpreters/Streaming/CachedBlockMetrics.cpp:10]: Do we need serialize the metrics ? Why

File: src/Processors/Transforms/Streaming/ChangelogConvertTransform.cpp [hunk 4]
```diff
@@ -427,7 +440,9 @@ void ChangelogConvertTransform::recover(CheckpointContextPtr ckpt_ctx)
             rb);
 
         DB::readIntBinary(late_rows, rb);
-        cached_block_metrics.deserialize(rb);
+
+        if (version_ <= CachedBlockMetrics::HAS_STATE_MAX_VERSION)
```
Comment (chenziliang) [inline src/Processors/Transforms/Streaming/ChangelogConvertTransform.cpp]: `CachedBlockMetrics::STATE_MAX_VERSION` ?

File: src/Interpreters/Streaming/CachedBlockMetrics.h [hunk 5]
```diff
@@ -11,25 +13,30 @@ namespace Streaming
...
-            gced_blocks);
+            gced_blocks,
+            total_metadata_bytes,
+            total_data_bytes);
     }
 
-    void serialize(WriteBuffer & wb) const;
-    void deserialize(ReadBuffer & rb);
+    /// [Legacy] We don't need to serialize this anymore on new impl, since the metrics is volated. will update it back during recover 
+    static constexpr VersionType HAS_STATE_MAX_VERSION = 1;
```
Comment (chenziliang) [inline src/Interpreters/Streaming/CachedBlockMetrics.h]: What's `HAS_STATE_MAX_VERSION`, didn't get what it means ? It is for metrics ?

File: src/Core/Block.cpp (hunk 1 above)
Comment (yl-lisen) [reply src/Core/Block.cpp:411]: yes, but it's difficult to collect them. every ColumnWithTypeAndName are different allocate. maybe.

File: src/Interpreters/Streaming/Aggregator.cpp (hunk 2 above)
Comment (yl-lisen) [reply src/Interpreters/Streaming/Aggregator.cpp]: I have log it, looks good.  `0 ms`.
```c++
LOG_INFO(
        log,
        "Took {} milliseconds to log metrics. Aggregated data metrics: {}",
        end - start,
        aggregated_data_metrics.string());
```

File: src/Interpreters/Streaming/CachedBlockMetrics.cpp (hunk 3 above)
Comment (yl-lisen) [reply src/Interpreters/Streaming/CachedBlockMetrics.cpp:10]: Before release new version, current version still is 1, so it is required for CI test on version 1

File: src/Interpreters/Streaming/CachedBlockMetrics.h (hunk 5 above)
Comment (yl-lisen) [reply src/Interpreters/Streaming/CachedBlockMetrics.h]: It is used to indicate the max version that needs to continue to do state serialization/deserialization.
use `SERDE_REQUIRED_MAX_VERSION` ?

File: tests/stream/test_stream_smoke/0013_changelog_stream13.yaml [hunk 6]
```diff
@@ -294,9 +295,9 @@ tests:
           from v_12180413 settings checkpoint_interval=1;
       - client: python
         query_type: table
-        depends_on_stream: v_12180413
+        depends_on_stream: 12180413213
```
Comment (yokofly) [inline tests/stream/test_stream_smoke/0013_changelog_stream13.yaml]: is this test flaky? why need change here..?

Comment (yokofly) [reply tests/stream/test_stream_smoke/0013_changelog_stream13.yaml]:
```suggestion
        depends_on: 12180413213
```

Comment (yl-lisen) [reply tests/stream/test_stream_smoke/0013_changelog_stream13.yaml]: > is this test flaky? why need change here..?

yes, it's flaky one

**Review by qijun-niu-timeplus — APPROVED** [review]

File: src/Interpreters/Streaming/Aggregator.cpp (hunk 2 above)
Comment (chenziliang) [reply src/Interpreters/Streaming/Aggregator.cpp]: let's track microseconds level probably.

Comment (yl-lisen) [reply src/Interpreters/Streaming/Aggregator.cpp]: > let's track microseconds level probably.

make sense, I can modify it in another pr

**Pull request merged**